ext4/dmverity/dmverity.go

```diff
@@ -69,20 +73,19 @@ type VerityInfo struct {
 	Version       uint32
 }
 
-// MerkleTree constructs dm-verity hash-tree for a given byte array with a fixed salt (0-byte) and algorithm (sha256).
-func MerkleTree(data []byte) ([]byte, error) {
+// MerkleTree constructs dm-verity hash-tree for a given bufio.Reader with a fixed salt (0-byte) and algorithm (sha256).
+func MerkleTree(br *bufio.Reader) ([]byte, error) {
```

Comment:
Why does this have to take a bufio.Reader explicitly and not just io.Reader?

Reply:
did I misinterpret your comment here: https://github.com/microsoft/hcsshim/pull/1209#discussion_r737947642? 😄

Reply:
or maybe `io.Reader` as param and a type check for `bufio.Reader` prior to initializing a new reader is the way to go here? 🤔

Reply:
I think just ensuring our callers of this pass a bufio.Reader is fine, it ensures someone could make their own Reader with their own custom logic (buffering or otherwise) if need be.

Reply:
Sorry if the above is confusing, I mean anywhere we call `MerkleTree` in our code just ensure we're passing a `bufio.Reader`, but I'd like the function signature to just take an `io.Reader` and not have any casting logic inside that cares about what it is.

Reply:
ah, gocha gocha.